homeassistant/components/tag/__init__.py

```diff
@@ -77,6 +100,18 @@ async def _process_create_data(self, data: dict) -> dict:
         # make last_scanned JSON serializeable
         if LAST_SCANNED in data:
             data[LAST_SCANNED] = data[LAST_SCANNED].isoformat()
+
+        # Create entity in entity_registry when creating the tag
+        # This is done early to store name only once in entity registry
+        self.entity_registry.async_get_or_create(
+            DOMAIN,
+            DOMAIN,
+            data[TAG_ID],
+            original_name=data.get(CONF_NAME, DEFAULT_NAME),
```

Comment:
Tags never have an original name. To simplify the logic, I think we should only use the name set by user.
An original name would be meaningful if `async_scan_tag` was extended with a name.

Reply:
In that case a tag's original name could only be the tag id or `DEFAULT_NAME` and don't think neccessarily any of those makes sense. So I'll remove the use of `original_name` all together and only use `name`.